dydx/dydxViews/dydxViews/_v4/TakeProfitStopLoss/Components/dydxCustomAmountViewModel.swift

```diff
@@ -0,0 +1,165 @@
+            self.onEdited?(isOn ? "\(minimumValue ?? 0)" : nil)
+        }
+        .createView()
+    }
+
+    private var input: AnyView? {
+        guard isOn else { return nil }
+        return super.createView()
+            .makeInput()
+            .wrappedInAnyView()
+    }
+
+    private lazy var slider: UISlider = {
+        let slider = UISlider(frame: .zero)
+        slider.value = 0
+        slider.thumbTintColor = ThemeColor.SemanticColor.textSecondary.uiColor
+        slider.minimumTrackTintColor = .clear
+        slider.maximumTrackTintColor = .clear
+        slider.addTarget(
+            self,
+            action: #selector(self.sliderValueChanged),
+            for: .valueChanged
+        )
+
+        return slider
+    }()
+
+    private var sliderBackground: some View {
+        GeometryReader { geometry in
+            let tickWidth = 1.5
+            let tickSpacing = 5.0
+            let numTicks = (geometry.size.width - tickSpacing * 2) / (tickWidth + tickSpacing)
+            return HStack(spacing: tickSpacing) {
+                Spacer()
+                HStack(spacing: tickSpacing) {
+                    ForEach(0..<Int(numTicks), id: \.self) { _ in
+                        RoundedRectangle(cornerRadius: tickWidth/2)
+                            .themeColor(background: .textTertiary)
+                            .frame(width: tickWidth, height: 8)
+                    }
+                }
+                Spacer()
+            }
+            .wrappedInAnyView()
+        }
+        .frame(height: 8)
+    }
+
+    private var sliderCompositeView: some View {
+        return ZStack(alignment: .center) {
+            self.sliderBackground
+            self.slider.swiftUIView
+        }
+    }
+
+    public static var previewValue: dydxTriggerPriceInputViewModel = {
+        let vm = dydxTriggerPriceInputViewModel(triggerType: .takeProfit)
+        return vm
+    }()
+
+    public override func createView(parentStyle: ThemeStyle = ThemeStyle.defaultStyle, styleKey: String? = nil) -> PlatformView {
+        PlatformView(viewModel: self, parentStyle: parentStyle, styleKey: styleKey) { [weak self] _  in
+            // if min == max, there is no need for a custom amount since there can only be one value
+            guard let self = self, minimumValue != maximumValue else { return PlatformView.emptyView.wrappedInAnyView() }
+            return VStack(spacing: 15) {
+                self.onOffSwitch
+                HStack(alignment: .center, spacing: 20) {
+                    if self.isOn {
+                        self.sliderCompositeView
+                        self.input
+                    }
+                }
+            }
+            .wrappedInAnyView()
+        }
+    }
+
+    @objc private func sliderValueChanged(sender: UISlider) {
+        guard let stepSizeDecimals else { return }
+        let roundedValue = dydxFormatter.shared.raw(number: NSNumber(value: sender.value), digits: stepSizeDecimals)
```

Comment:
Markets like Shiba have stepSize like 100, 1000, 10000, so we'd need to round the size with something like

`size =  Int(size / stepSize) * stepSize`

Reply:
ah, i thought the raw function did that, looks like it does in some implementations, I will update in this PR